lib/Rx/Observable.php

```diff
@@ -504,6 +505,19 @@ public function skipUntil(ObservableInterface $other)
     }
 
     /**
+     * Returns an observable sequence that produces a value after dueTime has elapsed and then after each period.
+     *
+     * @param $dueTime
+     * @param int|null $period
+     * @param SchedulerInterface $scheduler
+     * @return TimerObservable
+     */
+    public static function timer($dueTime, $period = 0, SchedulerInterface $scheduler = null)
```

Comment:
Thinking about the API here. What we currently have seems more like `Interval` to me. [1] `Timer` [2] is emitting one item after an interval, though RxJS has a version that seems to be a combination of both (e.g. an interval that starts _after_ the timer has fired).

Let's add two operators, one `Timer` and one `Interval` that behaves like described and then add some helper methods? E.g.

```
timer($dueTime, $scheduler);
interval($period, $scheduler);
intervalStartingAfter($dueTime, $period, $scheduler); // Can be implemented in terms of the other two?
```

[1]: http://reactivex.io/documentation/operators/interval.html
[2]: http://reactivex.io/documentation/operators/timer.html